server/index.js

```diff
@@ -23,31 +24,34 @@ app.use((req, res, next) => {
   next(createError(404));
 });
 
-const connections = [null, null];
+const games = {};
 
 io.on('connection', (socket) => {
-  let playerIndex = -1;
-  connections.forEach((value, index) => {
-    if (value === null) {
-      playerIndex = index;
+  socket.on('joinGame', (playerInfo) => {
+    const availableGame = Object.values(games);
+    const openGame = availableGame.find(game => game.freeToJoin === true);
+    if (openGame) {
+      openGame.pacmanTwo = playerInfo;
+      openGame.freeToJoin = false;
+      // eslint-disable-next-line no-param-reassign
+      socket.room = openGame.roomId;
+      socket.join(openGame.roomId);
+    } else {
+      const newGame = gameController(playerInfo);
+      games.roomId = newGame;
+      // eslint-disable-next-line no-param-reassign
+      socket.room = newGame.roomId;
+      socket.join(newGame.roomId);
     }
   });
-  socket.emit('player-number', playerIndex);
-  if (playerIndex === -1) return;
-  connections[playerIndex] = socket;
-  socket.broadcast.emit('player-connect', playerIndex);
-  socket.on('actuate', (data) => {
-    const { pacman } = data;
-    const move = {
-      playerIndex,
-      pacman,
-    };
-    socket.on('disconnect', () => {
-      connections[playerIndex] = null;
-    });
-    socket.broadcast.emit('move', move);
+
+  socket.on('disconnect', () => {
+    delete games[socket.room];
+    socket.leave(socket.room);
```

Comment:
Refactor this to how users have a separate folder. Or else no point of having structure.

Reply:
Okay